swift-extensions/UIPickerExtensions.swift

```diff
@@ -0,0 +1,59 @@
+import UIKit
+import TRIKOT_FRAMEWORK_NAME
+
+extension UIPickerView {
+    private struct AssociatedKeys {
+        static var elementsKey = UnsafeMutablePointer<Int8>.allocate(capacity: 1)
+    }
+
+    private var elements: [String]? {
+        get {
+             return objc_getAssociatedObject(self, AssociatedKeys.elementsKey) as? [String]
+        }
+        set {
+            objc_setAssociatedObject(self, AssociatedKeys.elementsKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN)
+        }
+    }
+    
+    public var pickerViewModel: PickerViewModel? {
+        get { return trikotViewModel() }
+        set(value) {
+            viewModel = value
+            
+            guard let pickerViewModel = value else { return }
+
+            self.delegate = self
+            observe(pickerViewModel.selectedElementIndex) { [weak self] (value: Int) in
+                self?.selectRow(value, inComponent: 0, animated: false)
+            }
+
+            observe(pickerViewModel.elements) { [weak self] (value: [PickerItemViewModel]) in
+                self?.elements = value.map {
+                    $0.displayName
+                }
+            }
+            
+            bind(pickerViewModel.enabled, \UIPickerView.isUserInteractionEnabled)
+        }
+    }
+}
+
+extension UIPickerView : UIPickerViewDelegate {
+    public func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        self.pickerViewModel?.setSelectedElementIndex(index: Int32(row))
```

Comment:
You can probably use UIControlEvent for that matter instead of declaring global conrormance to UIPickerViewDelegate

Reply:
Did that work ? I'm seeing that UIPickerView is not a UIControl 😭

Reply:
true!